### ISSUE TITLE
redraw() buggy

### ISSUE THREAD

**Issue by iniatse** [highcharts/highcharts#14163]
#### Expected behaviour
The chart should be redrawn on `redraw()`. I'm pretty sure that there is a clear consensus that a chart should be redrawn when `redraw()` gets executed regardless wheter a new series was added or not.

#### Actual behaviour
The chart gets not redrawn if no new series gets added. 

#### Live demo with steps to reproduce
New series --> `redraw()` works: http://jsfiddle.net/2aznmk9d/
No new series --> `redraw()` buggy: http://jsfiddle.net/rzLo3saq/

#### Product version
`<script src="https://code.highcharts.com/highcharts.js"></script>`

#### Affected browser(s)
All

**Comment by pawelfus** [comment]
Hi @iniatse 

Thanks for reporting the issue, but I don't quite understand the problem. Could you elaborate?

Just to be on the same page: `chart.redraw()` is redrawing the chart if there was any other method called, that set `redraw` argument to `false` (like in your first demo). Additionally, `chart.redraw()` is smart enough to redraw only components that need redrawing. For example: calling `point.update({ y: 0 });` where extremes are: `<-5, 5>` won't redraw yAxis because extremes didn't change.

**Comment by iniatse** [comment]
> Thanks for reporting the issue, but I don't quite understand the problem. Could you elaborate?

I would like to reanimate the chart in even in case that there are no changes to the chart. 

For example lets say you have a slider with multiple highcharts in it (each slide has one highcharts in it) and you want to reanimate the chart each time you move to a new slide. I found some reanimate() functions in your forum, but they do not seem production ready, because they do not work for all types of charts and furthermore do not provide any level of abstraction. I don't like to deal with third party API internals in order to achieve such basic things.

Furthermore I recently found a post where it is suggest do destroy and recreate the chart. Bug why so complicated?

I think everybody expects the chart to be redrawn regardless of if there were changes to it or not when calling `redraw()` and even if you do not agree with that, then there should be at least an option that can be passed as an argument to the ``redraw()` which enables a redraw (a complete redraw of the chart, like as if it was destroyed and recreated) regardless of if there were changes to the chart or not.

**Comment by pawelfus** [comment]
Thank you for the details!

> For example lets say you have a slider with multiple highcharts in it (each slide has one highcharts in it) and you want to reanimate the chart each time you move to a new slide.

Did you consider creating chart when you move to a new slide? There's no point in animating all charts in the background, when your user does not see the chart. I think it's not optimal.

> Furthermore I recently found a post where it is suggest do destroy and recreate the chart. Bug why so complicated?

That's how you can achieve this right now. There's no "reanimate" method. It's not complicated if you wrap this is a function, for example: `createChart() { if (chart) { chart.destroy() } Highcharts.chart(...);`. That solution does not have perfect performance though.

> I think everybody expects the chart to be redrawn regardless of if there were changes to it or not when calling redraw() and even if you do not agree with that, then there should be at least an option that can be passed as an argument to the ``redraw()` which enables a redraw (a complete redraw of the chart, like as if it was destroyed and recreated — the animation should be replayed) regardless of if there were changes to the chart or not.

You are right, I disagree with you. Imagine calling `chart.series[0].addPoint(point, redraw=false)`, then calling `chart.redraw()` and as a result everything on your chart reanimates. When you present something new on the chart, you want your users to focus on that particular thing, and this is when animation helps. [`Chart.redraw()`](https://api.highcharts.com/class-reference/Highcharts.Chart#redraw) was created to change a chart with multiple settings (e.g. `chart.addAxis() + chart.addSeries()`) and redrawing chart only once, to improve performance.

Anyway, it's not a bug, this is expected behaviour. If you think that `chart.reanimate()` or something similar is needed, could you create another ticket and follow our `Feature request` template? Thanks!

**Comment by iniatse** [comment]
> Did you consider creating chart when you move to a new slide? There's no point in animating all charts in the background, when your user does not see the chart. I think it's not optimal.

I never said I want to reanimate all charts within all slides when moving to a new slide. I just want to reanimate the now visible one after the move.

> That's how you can achieve this right now. There's no "reanimate" method. It's not complicated if you wrap this is a function, for example: createChart() { if (chart) { chart.destroy() } Highcharts.chart(...);. That solution does not have perfect performance though.

But this is just a workaround, but an acceptable one. Why not just add a native `reanimate()` to the highcharts API?

> You are right, I disagree with you...

I understand, but why not then add a new function to the API called for example `reanimate()`?

In conclusion, yes I think `reanimate()` is needed. Thx for the answers, I'll open another ticket.

**Comment by iniatse** [comment]
Imagine it is much easier and generic to just query wheter or not there's a highcharts within a slide or not and if so reanimate it via `reanimate()` then the other option with destroying and recreating the chart.

**Comment by pawelfus** [comment]
> I never said I want to reanimate all charts within all slides when moving to a new slide. I just want to reanimate the now visible one after the move.

Yes, sorry for the confusion, let me rephrase. When you create a chart, animation is started. You want to reanimate charts later again. Instead of reanimating chart, you can create chart when needed (in the use case above it is when users gets the slide visible). Of course, it's not perfect, because if user revisits the slide, reanimating is needed anyway..

> Why not just add a native `reanimate()` to the highcharts API?

No reason, that's why I asked you to create a ticket for that :) I think it's a good idea, we can make other devs work easier (that's the point of using libraries, right?). For example reanimating chart when user scrolls the website to see a chart. Then reanimating this again when user scrolls back to the chart. Legend and axes remain in place and only series (and annotations I guess) are animating again.

**Comment by iniatse** [comment]
I added a feature request: [#14168](https://github.com/highcharts/highcharts/issues/14168)

**Comment by pawelfus** [comment]
Thanks! Let me close this ticket then
